stubs/WebOb/webob/dec.pyi

```diff
@@ -1,49 +1,196 @@
+        args: tuple[()] = (),
+        kwargs: None = None,
+        middleware_wraps: None = None,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: wsgify[_RequestT_contra, []],
+        func: _RequestHandler[_RequestT_contra, []] | None,
+        RequestClass: type[_RequestT_contra],
+        args: tuple[()] = (),
+        kwargs: None = None,
+        middleware_wraps: None = None,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: wsgify[_RequestT_contra, []],
+        func: _RequestHandler[_RequestT_contra, []] | None = None,
+        *,
+        RequestClass: type[_RequestT_contra],
+        args: tuple[()] = (),
+        kwargs: None = None,
+        middleware_wraps: None = None,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: wsgify[Request, [_AppT_contra]],
+        func: _Middleware[Request, _AppT_contra, []] | None = None,
+        RequestClass: None = None,
+        args: tuple[()] = (),
+        kwargs: None = None,
+        *,
+        middleware_wraps: _AppT_contra,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: wsgify[_RequestT_contra, [_AppT_contra]],
+        func: _Middleware[_RequestT_contra, _AppT_contra, []] | None,
+        RequestClass: type[_RequestT_contra],
+        args: tuple[()] = (),
+        kwargs: None = None,
+        *,
+        middleware_wraps: _AppT_contra,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: wsgify[_RequestT_contra, [_AppT_contra]],
+        func: _Middleware[_RequestT_contra, _AppT_contra, []] | None = None,
+        *,
+        RequestClass: type[_RequestT_contra],
+        args: tuple[()] = (),
+        kwargs: None = None,
+        middleware_wraps: _AppT_contra,
+    ) -> None: ...
+    @overload
+    def __get__(self, obj: None, type: type[_S]) -> _unbound_wsgify[_RequestT_contra, _P, _S]: ...
+    @overload
+    def __get__(self, obj: object, type: type | None = None) -> Self: ...
+    @overload
+    def __call__(self, env: WSGIEnvironment, /, start_response: StartResponse) -> Iterable[bytes]: ...
+    @overload
+    def __call__(self, func: _RequestHandler[_RequestT_contra, _P], /) -> Self: ...
+    @overload
+    def __call__(self, req: _RequestT_contra) -> _AnyResponse: ...
+    @overload
+    def __call__(self, req: _RequestT_contra, *args: _P.args, **kw: _P.kwargs) -> _AnyResponse: ...
+    def get(self, url: str, **kw: Any) -> _AnyResponse: ...
+    def post(
+        self, url: str, POST: str | bytes | Mapping[Any, Any] | Mapping[Any, list[Any] | tuple[Any, ...]] | None = None, **kw: Any
+    ) -> _AnyResponse: ...
+    def request(self, url: str, **kw: Any) -> _AnyResponse: ...
+    def call_func(self, req: _RequestT_contra, *args: _P.args, **kwargs: _P.kwargs) -> _AnyResponse: ...
+    # technically this could bind different type vars, but we disallow it for safety
+    def clone(self, func: _RequestHandler[_RequestT_contra, _P] | None = None, **kw: Never) -> Self: ...
```

Comment:
Why not omit `**kw`? That probably produces better errors if users try to pass args.

Reply:
That's fair, I think I mostly did this to avoid the stubtest error and `Never` in parameters produces far better error messages than in return types.